### DISCUSSION TITLE
Synchronize modes: local, cloud or none of both - no real synchronization with the office?!

### DISCUSSION THREAD

**Discussion by BoswachterMarc** [opengisch/QField#2461] [Q&A]
Let me use the following example:

1. In the field I have a colleague who makes changes to the project in QField and pushes the changes to the QField cloud
2. In the office, I make changes to the project and I synchronize the project with the QField Sync plug-in 4.0.0-BETA15 (in QGIS 3.16.3)
3. When synchronizing in the office, I can prefer the local mode, the cloud mode, or the 'none of both' mode (see printscreen)
**4. I try the none of both mode because, I do not prefer the cloud mode (changes of my colleague), nor the local mode (the project I changed on my pc).**
5. If I prefer the cloud mode, I will lose the changes I made in the office.
6. If I prefer the local mode, I will lose the changes my colleague made in the field
**7. After using the none of both mode, I don't see the changes from my colleague in the field in the office on my pc (although I see the layer change/new value in the dashboard of the QFieldCloud website - see printscreen)...**
8. Then I use the local mode (imagine I didn't know my colleague was in the field) but I lost the changes my colleague made in the field
9. Imagine I knew my colleague was in the field, then I preferred the cloud mode (my colleague pushed the changes to the cloud) but I last the changes I made in the office...

In a second attempt, I made a change in the QField app and I synchronized (in stead of pushing the changes to the cloud). I also changed a feature in the project on my pc. Again I synchronized preferring 'none of both'. The result is that the changes that were made in the QField app are not synchronized (although the QFieldSync plug-in says the project was synchronized after using the 'none of both' method), the changes that I made on my pc are still there.

In other words, there is something I don't get :-(
**When I use the 'none of both' mode, I would expect that both changes will be part of the 'new' synchronized project. Having to choose between the local (office) or the cloud (field) version is resulting in loosing or the office, or the field changes?!** In other words, then there is no synchronization with the office at all...
Please help. Thanks.

![Schermafdruk_2021-11-03_18-49-36](https://user-images.githubusercontent.com/57868975/140182291-ca052df7-4180-40aa-9fc1-f8c164d6d9d0.png)
![Schermafdruk_2021-11-03_21-03-08](https://user-images.githubusercontent.com/57868975/140184712-b61ee927-a73a-4fed-b843-dc27b13fa02b.png)
![Schermafdruk_2021-11-03_21-19-37](https://user-images.githubusercontent.com/57868975/140187583-8f3e9aab-8b8f-40bd-9a03-65d987c6c542.png)
![Schermafdruk_2021-11-03_21-20-17](https://user-images.githubusercontent.com/57868975/140187841-8082ec8d-7cd3-4f21-b102-e8132ee290c9.png)

**Comment by BoswachterMarc** [comment]
I have been using QFieldCloud quite intensive the last 2 weeks. Synchronisation is always smooth (from both sides, I mean on mobile and on pc), I never had sync errors :-)

1.  Nevertheless, uploading or downloading changes doesn't work at once; I mean there is a kind of time-out when you want to synchronize different gpkg and qgs files. So I sync in two times (changed files not all checked for synchronization at once > see printscreen)
2. Before working on the project at the office, I always check if field collaborators made any changes. If not, I start working on the project or make some changes in the office. If there are changes from collaborators, I first sync from cloud to local project on the pc in the office.  I am just afraid that a certain moment, changes will happen simultaneous and that one of both parties will lose his changes... This seems to me to be an important point of attention (or am I overlooking something?).

Apart from the above, it's a pleasure to be able to work with qgis and the mobile applications and synchronization!

![image](https://user-images.githubusercontent.com/57868975/141697759-0168f27f-4f19-4b8d-b4fd-1085e3385352.png)

**Reply by suricactus** [reply]
Hey, @BoswachterMarc are you still experiencing the problems you mentioned here? If no, can we close this discussion and if you encounter new problems you can open a new issue or discussion?

**Reply by suricactus** [reply]
@BoswachterMarc ping. I think if you upgrade to QFieldSync 4.0.0. BETA20 most of your timeout issue should be gone?

**Reply by mygiano** [reply]
me too, is there development to add synchronize mode that update the data from QGIS desktop from office and field only the changed and new data??
Maybe 3 options mode,
local, cloud, and newly changes

**Reply by albinepro** [reply]
I don't entirely get the logic of sync too. I understand that everything pushed from QField mobile app will be logged and changed feature by feature on the cloud. But on desktop application this will not happen - in the [docs](https://docs.qfield.org/reference/qfieldcloud/projects/) there's written `Changes to vector layers done in QGIS will not appear here.`
So changes made on desktop QGIS will be moved by **overwriting** the cloud files - this is concern of local file storage (spatialite, gpkg, shp...) not server PostGIS storage.

**Comment by marioba** [comment]
In order to provide a clearer and more manageable discussion forum and to have all discussions about QField and related products (i.e. QFieldCloud and QFieldSync) in a single place, we transferred all the discussions from the QFieldCloud to the [QField repository](https://github.com/opengisch/QField/discussions).

**Comment by suricactus** [comment]
Closing due to lack of feedback. The problems should have been fixed by now. Please open a new issue in case you still experience issue.
